main.js

```diff
@@ -52,10 +52,39 @@ function RoundToHundredths(x) {
   return Math.floor(x * 100 + 0.5) / 100;
 }
 
-function HandleNode(svgNode, scaleX, scaleY, translateX, translateY) {
+// |fillString| is expected to be "#RRGGBB" or "#RGB"].
+function PathColorFromFill(fillString) {
+  if(fillString.length === 4) {
+    // Color in form of #RGB so let's turn that to #RRGGBB.
+    fillString = `#${fillString[1]}${fillString[1]}${fillString[2]}${fillString[2]}${fillString[3]}${fillString[3]}`.toUpperCase();
+  }
+
+  const r = fillString.substr(1,2);
+  const g = fillString.substr(3,2);
+  const b = fillString.substr(5,2);
+  
+  return `PATH_COLOR_ARGB, 0xFF, 0x${r}, 0x${g}, 0x${b},\n`;
+}
+
+function HandleNode(svgNode, scaleX, scaleY, translateX, translateY, preserveFill) {
   var output = '';
   for (var idx = 0; idx < svgNode.children.length; ++idx) {
     var svgElement = svgNode.children[idx];
+
+    if (preserveFill) {
+      output += "NEW_PATH,\n";
```

Comment:
you don't need or want NEW_PATH for the first path. Also I don't think you want it unless there's a fill, i.e. this should with L83

Reply:
Done so. Wondering if adding `NEW_PATH` only if there's an element that has fill causes an issue.

Imagine the following SVG:
```
<svg>
  <path d="path1" fill="#000"/>
  <path d="path2"/>
</svg>

Without outputting `NEW_PATH` for each elem, won't the fill from "path1" bleed and be applied also to "path2"? Lmk what you think

Reply:
@evanstade ping 😁

Reply:
Thanks for the ping. I think you are right.